File: matrix-sdk-android/src/main/java/org/commonmark/ext/maths/internal/MathsHtmlNodeRenderer.java
/*
 * Copyright (c) 2020 New Vector Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.commonmark.ext.maths.internal;

import org.commonmark.ext.maths.DisplayMaths;
import org.commonmark.node.Node;
import org.commonmark.node.Text;
import org.commonmark.renderer.html.HtmlNodeRendererContext;
import org.commonmark.renderer.html.HtmlWriter;

import java.util.Collections;
import java.util.Map;

public class MathsHtmlNodeRenderer extends MathsNodeRenderer {
    private final HtmlNodeRendererContext context;
    private final HtmlWriter html;

    public MathsHtmlNodeRenderer(HtmlNodeRendererContext context) {
        this.context = context;
        this.html = context.getWriter();
    }

    @Override
    public void render(Node node) {
        boolean display = node.getClass() == DisplayMaths.class;
        Node contents = node.getFirstChild(); // should be the only child
        String latex = ((Text) contents).getLiteral();
        Map<String, String> attributes = context.extendAttributes(node, display ? "div" : "span", Collections.<String, String>singletonMap("data-mx-maths",
                latex));
        html.tag(display ? "div" : "span", attributes);
        html.tag("code");
        context.render(contents);
        html.tag("/code");
        html.tag(display ? "/div" : "/span");
    }
}

File: matrix-sdk-android/src/main/java/org/commonmark/ext/maths/internal/MathsHtmlNodeRenderer.kt
/*
 * Copyright 2025 New Vector Ltd.
 * Copyright 2021 The Matrix.org Foundation C.I.C.
 *
 * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
 * Please see LICENSE files in the repository root for full details.
 */
package org.commonmark.ext.maths.internal

import org.commonmark.ext.maths.DisplayMaths
import org.commonmark.node.Node
import org.commonmark.node.Text
import org.commonmark.renderer.html.HtmlNodeRendererContext
import org.commonmark.renderer.html.HtmlWriter
import java.util.Collections

internal class MathsHtmlNodeRenderer(private val context: HtmlNodeRendererContext) : MathsNodeRenderer() {
    private val html: HtmlWriter = context.writer
    override fun render(node: Node) {
        val display = node.javaClass == DisplayMaths::class.java
        val contents = node.firstChild // should be the only child
        val latex = (contents as Text).literal
        val attributes = context.extendAttributes(
                node, if (display) "div" else "span", Collections.singletonMap(
                "data-mx-maths",
                latex
        )
        )
        html.tag(if (display) "div" else "span", attributes)
        html.tag("code")
        context.render(contents)
        html.tag("/code")
        html.tag(if (display) "/div" else "/span")
    }
}
